packages/flutter/lib/src/cupertino/refresh.dart

```diff
@@ -359,43 +357,53 @@ class CupertinoSliverRefreshControl extends StatefulWidget {
     return state.refreshState;
   }
 
-  /// Builds a simple refresh indicator that fades in a bottom aligned down
-  /// arrow before the refresh is triggered, a [CupertinoActivityIndicator]
-  /// during the refresh and fades the [CupertinoActivityIndicator] away when
-  /// the refresh is done.
-  static Widget buildSimpleRefreshIndicator(
+  /// Builds a refresh indicator that reflects the standard iOS pull-to-refresh
+  /// behaviour. Specifically, this entails presenting an activity indicator that
+  /// changes depending on the current refreshState. As the user initially drags
+  /// down, the indicator will gradually reveal individual ticks until the refresh
+  /// becomes armed. At this point, the animated activity indicator will begin rotating.
+  /// Once the refresh has completed, the activity indicator shrinks away as the
+  /// space allocation animated back to closed.
+  static Widget buildRefreshIndicator(
     BuildContext context,
     RefreshIndicatorMode refreshState,
     double pulledExtent,
     double refreshTriggerPullDistance,
     double refreshIndicatorExtent,
   ) {
-    const Curve opacityCurve = Interval(0.4, 0.8, curve: Curves.easeInOut);
+    final double percentageComplete = min(pulledExtent / refreshTriggerPullDistance, 1.0);
     return Align(
-      alignment: Alignment.bottomCenter,
+      alignment: Alignment.topCenter,
       child: Padding(
-        padding: const EdgeInsets.only(bottom: 16.0),
-        child: refreshState == RefreshIndicatorMode.drag
-            ? Opacity(
-                opacity: opacityCurve.transform(
-                  min(pulledExtent / refreshTriggerPullDistance, 1.0)
-                ),
-                child: Icon(
-                  CupertinoIcons.down_arrow,
-                  color: CupertinoDynamicColor.resolve(CupertinoColors.inactiveGray, context),
-                  size: 36.0,
-                ),
-              )
-            : Opacity(
-                opacity: opacityCurve.transform(
-                  min(pulledExtent / refreshIndicatorExtent, 1.0)
-                ),
-                child: const CupertinoActivityIndicator(radius: 14.0),
-              ),
+        padding: const EdgeInsets.only(top: 16.0),
+        child: _buildIndicatorForRefreshState(refreshState, 14.0, percentageComplete),
       ),
     );
   }
 
+  static Widget _buildIndicatorForRefreshState(RefreshIndicatorMode refreshState, double radius, double percentageComplete) {
```

Comment:
Nit: I've been called out on my PRs before for using static methods that return widgets and told to just make a new StatelessWidget instead.  In this case I realize that `buildSimpleRefreshIndicator` was already guilty of this though :) Just something to keep in mind if you see a better way to do this, otherwise it's fine as-is.